### PR TITLE
Refactor PjRt tests to take advantage of dynamic libtpu loading

### PR DESCRIPTION
**Pull request by will-cromar**
With the dynamic loading of libtpu implemented in #3616, we can more easily launch processes from the Python test script itself.

- Create `test/pjrt/common.py` with utilities for launching multiprocess+multithreaded workloads
  - Set the `XRT_SHARD_WORLD_SIZE` based on the number of visible devices (unblocks. TPU v4)
- Remove run_all_chips.sh
- Symlink `args_parse.py` from parent folder to `test/pjrt`
- Simplify MNIST and ImageNet PjRt tests

Tested with 1 TPU chip, 4 TPU chips, and CPU on v2-8 and v4-8

Will send a separate PR to update tests in ml-testing-accelerators.

### CODE REVIEW

File: test/pjrt/common.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+def configure_tpu_topology(rank: int, processes: int, base_port=8476):
+  '''Set default TPU topology environment variables for a single TPU host.'''
+  ports = list(range(base_port, base_port + processes))
+  os.environ.setdefault('TPU_CHIPS_PER_PROCESS_BOUNDS', '1,1,1')
+  os.environ.setdefault('TPU_PROCESS_BOUNDS', '2,2,1')
+  os.environ.setdefault('TPU_PROCESS_ADDRESSES',
+                        ','.join(f'localhost:{port}' for port in ports))
+
+  os.environ.setdefault('TPU_VISIBLE_DEVICES', str(rank))
+  os.environ.setdefault('TPU_PROCESS_PORT', str(ports[rank]))
```
Comment (JackCaoG) [inline test/pjrt/common.py:28]: what is this `TPU_PROCESS_PORT` used for?

File: test/pjrt/common.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    kwargs: kwargs to pass to `fn`
+
+  Returns:
+    Dict of the form {process_rank: {thread_rank: return_value}}, where
+    return_value is the result of calling `fn`.
+  '''
+  if pjrt_device() == 'TPU':
+    processes = num_visible_tpu_chips()
+  else:
+    processes = 1
```
Comment (JackCaoG) [inline test/pjrt/common.py:78]: how many process are we going to have for multi gpu?

**Review by JackCaoG — COMMENTED** [review]
mostly lgtm

File: test/pjrt/common.py (hunk 2 above)
Comment (will-cromar) [reply test/pjrt/common.py:78]: One process per GPU will almost certainly be best option. I haven't tested GPU at all yet, though

File: test/pjrt/common.py (hunk 1 above)
Comment (will-cromar) [reply test/pjrt/common.py:28]: It's an internal setting in libtpu that's only used when there are multiple communicating processes per TPU host. It corresponds to one of the ports in `TPU_PROCESS_ADDRESSES`

**Review by JackCaoG — APPROVED** [review]
Thanks!

**Comment by JackCaoG** [comment]
Hopefully we can just use ddp for pjrt instead of introducing a new way for people to run pt/xla model. 😄

**Pull request merged**